lib/src/command/login.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import '../command.dart';
+import '../io.dart' as io;
+
+import '../tokens.dart';
+
+/// Handles the `login` pub command.
+class LoginCommand extends PubCommand {
+  @override
+  String get name => 'login';
+  @override
+  String get description => 'Log in to third-party pub server.';
+  @override
+  String get invocation => 'pub login <server>';
```

Comment:
It is not clear for me what the `<server>` value would be: a `http://fqn.domain/` or just a hostname? I assume the former, but it would be nice if the help message would clarify it.

Reply:
I will add it to the help (validation will step in otherwise on what to expect). Any suggestion on how the message should be displayed? 
`  String get description => 'Log in to third-party pub server. Specify server as a fully qualified URL.';

Reply:
How about the publish command? Because there you have the server option with help text 'The package server to which to upload this package.'

I don' think the description is a good place to explain how the server value should look?
Also the 'invocation' value, is this a place to explain more?